tests/workflows/builtin/test_heal.py

```diff
@@ -0,0 +1,82 @@
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import pytest
+
+from aria.workflows.builtin.heal import heal
+
+from . import (simple_context,
+               assert_node_install_operations,
+               assert_node_uninstall_operations)
+
+
+@pytest.fixture
+def context():
+    return simple_context()
+
+
+def test_heal_dependent_node(context):
+    heal_graph = heal(context=context, node_instance_id='dependent_node_instance')
+    uninstall_subgraph, install_subgraph = list(heal_graph.topological_order(reverse=True))
+    dependent_node_subgraph_uninstall, dependency_node_subgraph_uninstall = \
+        list(uninstall_subgraph.topological_order(reverse=True))
+    dependency_node_subgraph_install, dependent_node_subgraph_install = \
+        list(install_subgraph.topological_order(reverse=True))
+
+    dependent_node_uninstall_operations = \
```

Comment:
consider modifying heal workflow so that it doesnt create empty subgraphs when unnecessary